packages/system/src/conditional.tsx

```diff
@@ -0,0 +1,38 @@
+export type State = {
+  checked?: boolean;
+  focus?: boolean;
+  hover?: boolean;
+  disabled?: boolean;
+  error?: boolean;
+};
+
+const toVariantState = (variant: string, state: string) => {
+  return `${variant}.:${state}`;
+};
+
+/**
+ * function to create an array of variant and their conditional props
+ */
+export const conditional = (variant: string, state: State) => {
```

Comment:
Can you destruct `disabled` here to simplify the function? 
`(variant, { disabled = false , ... states})